pylint/utils/utils.py

```diff
@@ -17,13 +17,55 @@
 import sys
 import textwrap
 import tokenize
+from typing import (
+    TYPE_CHECKING,
+    List,
+    Optional,
+    Pattern,
+    Tuple,
+    TypeVar,
+    Union,
+    overload,
+)
 
 from astroid import Module, modutils
 
 from pylint.constants import PY_EXTS
 
+if sys.version_info >= (3, 8):
+    from typing import Literal
+else:
+    from typing_extensions import Literal
+
+if TYPE_CHECKING:
+    from pylint.checkers.base_checker import BaseChecker
+
 DEFAULT_LINE_LENGTH = 79
 
+# These are types used to overload get_global_option() and refer to the options type
+GLOBAL_OPTION_BOOL = Literal[
+    "ignore-mixin-members",
+    "suggestion-mode",
+    "analyse-fallback-blocks",
+    "allow-global-unused-variables",
+]
+GLOBAL_OPTION_INT = Literal["max-line-length", "docstring-min-length"]
+GLOBAL_OPTION_LIST = Literal["ignored-modules"]
+GLOBAL_OPTION_PATTERN = Literal[
+    "no-docstring-rgx", "dummy-variables-rgx", "ignored-argument-names"
+]
+GLOBAL_OPTION_TUPLE_INT = Literal["py-version"]
+GLOBAL_OPTION_NAMES = Union[
+    GLOBAL_OPTION_BOOL,
+    GLOBAL_OPTION_INT,
+    GLOBAL_OPTION_LIST,
+    GLOBAL_OPTION_PATTERN,
+    GLOBAL_OPTION_TUPLE_INT,
+]
+GLOBAL_OPTION_RETURN_TYPES = TypeVar(
+    "GLOBAL_OPTION_RETURN_TYPES", bool, int, List[str], Pattern, Tuple[int, ...]
+)
```

Comment:
Usually TypeVars are `PascalCase`, similar to class names or even just `T`.
What do you think about `T_GlobalOptionReturnTypes` (best of both worlds)?

Reply:
Sounds good!